src/include/migraphx/op/if_op.hpp

```diff
@@ -54,7 +54,8 @@ struct if_op
         if(not std::equal(
                out_shapes1.begin(), out_shapes1.end(), out_shapes0.begin(), out_shapes0.end()))
```

Comment:
From the ONNX spec, the output shapes of the submodules do not have to be the same. But, getting a "compatible" output shape for `If` requires dynamic shape, so might keep this check for now. 

I see that the code in `parse_if.cpp` will make the output shapes the same. I'm not sure exactly what changes will be made when this operator is updated to support dynamic shape. May want to discuss the design.

Reply:
My assumption is that they should be compatible out of the parse_if. I don't intend on changing the if_op yet since the spec mentioned static shapes. I'm assuming dynamic shapes would be equivalent on output of this?

Reply:
This looks to be one of the operators that can output a dynamic shape even if the inputs are static/fixed shapes. I'm basing this off of the spec writing:

"""For example, if in a model file, the first output of then_branch is typed float tensor with shape [2] and the first output of else_branch is another float tensor with shape [3], If's first output should have (a) no shape set, or (b) a shape of rank 1 with neither dim_value nor dim_param set, or (c) a shape of rank 1 with a unique dim_param. In contrast, the first output cannot have the shape [2] since [2] and [3] are not compatible."""

So the output shape lengths of the submodules can be different, but they should have the same number of dimensions and type. Per the example, the output of the `If` node could then have a shape of [2] or [3].

Reply:
More on this, I suppose at a later date we should add dynamic shapes to this operator. 

Does having a dynamic output on one branch result in the static shape taking precedence of what's the valid output? How would that generally work?